### PR TITLE
[linux] Fix compiler in native ARM host builds

### PR DESCRIPTION
**Pull request by fwcd**
#35084 introduced a regression for native ARM Linux builds where the toolchain would mistakenly try to reference a cross-compiler (e.g. `aarch64-linux-gnu-gcc`) despite building natively because vcpkg spells the architecture differently from what the system reports (`arm64` vs. `aarch64`).

This fixes the issue by checking whether the `CMAKE_HOST_SYSTEM_PROCESSOR` matches the target (in CMake's spelling of the architecture).

### CODE REVIEW

**Comment by dg0yt** [comment]
The direction is good but...
I think it is necessary to coordinate a review and update of the linux toolchain file to limit the number of expensive rebuilds which don't consider the whole crossbuild and native picture.

**Comment by fwcd** [comment]
I agree, I see this more as a small fix to the other PR. Since this check only applies to exactly this case (host is exactly of the same architecture as target), my hope is that this shouldn't regress anything new... a larger rewrite would be out-of-scope here

**Comment by dg0yt** [comment]
Even if it doesn't regress, it invalidates all cached artifacts for this toolchain. That's why I would prefer to see it bundled with other necessary fixes.

**Review by Cheney-W — DISMISSED** [review]

**Comment by Cheney-W** [comment]
Add `reviewed` tag for further team review.

**Review by BillyONeal — DISMISSED** [review]

File: scripts/toolchains/linux.cmake [hunk 1]
```diff
@@ -13,7 +13,7 @@ elseif(VCPKG_TARGET_ARCHITECTURE STREQUAL "x86")
    string(APPEND VCPKG_LINKER_FLAGS " -m32")
 elseif(VCPKG_TARGET_ARCHITECTURE STREQUAL "arm")
     set(CMAKE_SYSTEM_PROCESSOR armv7l CACHE STRING "")
-    if(CMAKE_HOST_SYSTEM_NAME STREQUAL "Linux")
+    if(CMAKE_HOST_SYSTEM_NAME STREQUAL "Linux" AND NOT CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL armv7l)
```
Comment (dg0yt) [inline scripts/toolchains/linux.cmake:16]:
```suggestion
    if(CMAKE_HOST_SYSTEM_NAME STREQUAL "Linux" AND NOT CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL "armv7l")

```

File: scripts/toolchains/linux.cmake [hunk 2]
```diff
@@ -30,7 +30,7 @@ elseif(VCPKG_TARGET_ARCHITECTURE STREQUAL "arm")
     endif()
 elseif(VCPKG_TARGET_ARCHITECTURE STREQUAL "arm64")
     set(CMAKE_SYSTEM_PROCESSOR aarch64 CACHE STRING "")
-    if(CMAKE_HOST_SYSTEM_NAME STREQUAL "Linux")
+    if(CMAKE_HOST_SYSTEM_NAME STREQUAL "Linux" AND NOT CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL aarch64)
```
Comment (dg0yt) [inline scripts/toolchains/linux.cmake:33]:
```suggestion
    if(CMAKE_HOST_SYSTEM_NAME STREQUAL "Linux" AND NOT CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL "aarch64")

```

**Pull request merged**